### PR TITLE
Fabo/ledger hid

### LINKED ISSUE #3247: Ledger on Windows
Allow Ledger to be used on Windows May 2019 Update
This requires switching to HID transportation

### PR DESCRIPTION
**Pull request by faboweb**
Closes #3247

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
Windows + HID not enabled warning

image

File: src/components/common/TmSessionHardware.vue [hunk 1]
```diff
@@ -6,11 +6,35 @@
...
         <HardwareState :loading="status === `connect` ? false : true">
-          <template v-if="status === `connect` || status === `detect`">
+          <template v-if="isWindows && !hasHIDEnabled">
+            Using a Ledger on Windows requires experimental HID support in your
+            browser.
+            <template v-if="isChrome">
+              <br />
+              <br />
+              <p>
+                Please copy this link into a new tab and 'enable' the shown
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]: can't just click the link with a blank target?

Comment (faboweb) [reply src/components/common/TmSessionHardware.vue]: First thing I tried ;P
Apparently doesn't work due to security rules

Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]:
```suggestion
                Please copy the link below into a new tab and 'enable' experimental web platform
```

File: src/components/common/TmSessionHardware.vue [hunk 2]
```diff
@@ -6,11 +6,35 @@
...
+                feature:
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]:
```suggestion
                features:
```

File: src/components/common/TmSessionHardware.vue [hunk 3]
```diff
@@ -61,6 +91,16 @@ export default {
...
       }[this.status]
+    },
+    isWindows() {
+      return this.navigator.platform.indexOf("Win") > -1
+    },
+    hasHIDEnabled() {
+      return !!this.navigator.hid
+    },
+    isChrome() {
+      const ua = navigator.userAgent.toLowerCase()
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]: no abbrv plz thx kk

File: src/components/common/TmSessionHardware.vue [hunk 4]
```diff
@@ -81,6 +121,12 @@ export default {
         sessionType: `ledger`,
         address: this.address
       })
+    },
+    onCopy() {
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]: we should make this a component that can be applied to any string.

**Pull request merged**